metricbeat/module/elasticsearch/elasticsearch.go

```diff
@@ -391,3 +391,24 @@ func getSettingGroup(allSettings common.MapStr, groupKey string) (common.MapStr,
 
 	return common.MapStr(v), nil
 }
+
+// IsCCRAvailable returns whether CCR is available or not, depending on the current license
+func IsCCRAvailable(http *helper.HTTP, resetURI string) (isAvailable bool, currentLicense string, err error) {
```

Comment:
I like both your ideas, actually. The version check should probably log+report at the same frequency as the license check. So how about I do this:

- Make a generic version check function in `elasticsearch.go`. In fact, I think there is one already.
- Make a generic license check function in `elasticsearch.go`.
- Move `isCCRAvailable` into `ccr.go`, since it's really only needed by that metricset. This function would check the license and the version, returning a boolean flag and an error message, if required.

WDYT?

Reply:
I like it :+1:

Reply:
Refactoring done in 49c422b31.